src/util/transactionUtils.ts

```diff
@@ -8523,6 +8542,384 @@ export class TransactionUtils {
+    let txn:
+      | PartialSdaExchangeTransaction
+      | UnconfirmedSdaExchangeTransaction
+      | ConfirmedSdaExchangeTransaction;
+    if (groupType == "partial") {
+      formattedTxn = await TransactionUtils.formatPartialTransaction(
+        transaction
+      );
+      txn = PartialTransaction.convertToSubClass(
+        PartialSdaExchangeTransaction,
+        formattedTxn
+      ) as PartialSdaExchangeTransaction;
+    } else if (groupType == "unconfirmed") {
+      formattedTxn = await TransactionUtils.formatUnconfirmedTransaction(
+        transaction
+      );
+      txn = UnconfirmedTransaction.convertToSubClass(
+        UnconfirmedSdaExchangeTransaction,
+        formattedTxn
+      ) as UnconfirmedSdaExchangeTransaction;
+    } else {
+      formattedTxn = await TransactionUtils.formatConfirmedTransaction(
+        transaction
+      );
+      txn = ConfirmedTransaction.convertToSubClass(
+        ConfirmedSdaExchangeTransaction,
+        formattedTxn
+      ) as ConfirmedSdaExchangeTransaction;
+    }
+
+    if (transaction.type === TransactionType.PLACE_SDA_EXCHANGE_OFFER) {
+      const sdaExchangeOfferTxn = transaction as PlaceSdaExchangeOfferTransaction;
+      for (let i = 0; i < sdaExchangeOfferTxn.offers.length; ++i) {
+        const tempSdaExchangeOffer = sdaExchangeOfferTxn.offers[i];
+
+        const assetIdGet = tempSdaExchangeOffer.mosaicIdGet.toHex();
+        const assetIdGive = tempSdaExchangeOffer.mosaicIdGive.toHex();
+        const amountGet = tempSdaExchangeOffer.mosaicAmountGet.compact();
+        const amountGive = tempSdaExchangeOffer.mosaicAmountGive.compact();
+        const duration = tempSdaExchangeOffer.duration.compact();
+
+      const newTxnSdaExchange: TxnSdaExchange = {
+        assetIdGet: assetIdGet,
+        assetIdGive: assetIdGive,
+        amountGet: amountGet,
+        amountGive: amountGive,
+        duration: duration,
+      };
+
+      try {
+        const assetInfoSdaGet = await TransactionUtils.getAssetInfo(assetIdGet);
+        const assetInfoSdaGive = await TransactionUtils.getAssetInfo(assetIdGive);
+
+        newTxnSdaExchange.amountGet = TransactionUtils.convertToExactAmount(
+          amountGet,
+          assetInfoSdaGet.divisibility
+        );
+        newTxnSdaExchange.amountGive = TransactionUtils.convertToExactAmount(
+          amountGive,
+          assetInfoSdaGive.divisibility
+        );
+
+        const assetSdaGetName = await TransactionUtils.getAssetName(assetIdGet);
+
+        if (assetSdaGetName.names.length) {
+          newTxnSdaExchange.assetSdaGetNamespace = assetSdaGetName.names[0].name;
+        }
+
+        const assetSdaGiveName = await TransactionUtils.getAssetName(assetIdGive);
+
+        if (assetSdaGiveName.names.length) {
+          newTxnSdaExchange.assetSdaGiveNamespace = assetSdaGiveName.names[0].name;
+        }
+      } catch (error) {}
+        txn.sdaExchange.push(newTxnSdaExchange);
+      }
+    }
+    return txn;
+  }
+
```

Comment:
missing `txn.sdaExchange.push(newTxnSdaExchange);` in formatUnconfirmedSdaExchangeTransaction, formatConfirmedSdaExchangeTransaction and formatPartialSdaExchangeTransaction. `newTxnSdaExchange` variable left unhandled